src/osd/SnapMapper.cc

```diff
@@ -992,120 +983,3 @@ void SnapMapper::Scrubber::run()
   mapit = ObjectMap::ObjectMapIterator();
 }
 #endif // !WITH_SEASTAR
-
-
-// -------------------------------------
-// legacy conversion/support
-
-string SnapMapper::get_legacy_prefix(snapid_t snap)
-{
-  ceph_assert(snap != CEPH_NOSNAP && snap != CEPH_SNAPDIR);
-  return fmt::sprintf("%s%.16X_",
-		      LEGACY_MAPPING_PREFIX,
-		      static_cast<uint64_t>(snap));
-}
-
-string SnapMapper::to_legacy_raw_key(
-  const pair<snapid_t, hobject_t> &in)
-{
-  return get_legacy_prefix(in.first) + shard_prefix + in.second.to_str();
-}
-
-bool SnapMapper::is_legacy_mapping(const string &to_test)
-{
-  return to_test.substr(0, LEGACY_MAPPING_PREFIX.size()) ==
-    LEGACY_MAPPING_PREFIX;
-}
-
-#ifndef WITH_SEASTAR
-/* Octopus modified the SnapMapper key format from
- *
- *  <LEGACY_MAPPING_PREFIX><snapid>_<shardid>_<hobject_t::to_str()>
- *
- * to
- *
- *  <MAPPING_PREFIX><pool>_<snapid>_<shardid>_<hobject_t::to_str()>
- *
- * We can't reconstruct the new key format just from the value since the
- * Mapping object contains an hobject rather than a ghobject. Instead,
- * we exploit the fact that the new format is identical starting at <snapid>.
- *
- * Note that the original version of this conversion introduced in 94ebe0ea
- * had a crucial bug which essentially destroyed legacy keys by mapping
- * them to
- *
- *  <MAPPING_PREFIX><poolid>_<snapid>_
- *
- * without the object-unique suffix.
- * See https://tracker.ceph.com/issues/56147
- */
-std::string SnapMapper::convert_legacy_key(
-  const std::string& old_key,
-  const ceph::buffer::list& value)
-{
-  auto old = from_raw(make_pair(old_key, value));
-  std::string object_suffix = old_key.substr(
-    SnapMapper::LEGACY_MAPPING_PREFIX.length());
-  return SnapMapper::MAPPING_PREFIX + std::to_string(old.second.pool)
-    + "_" + object_suffix;
-}
-
-int SnapMapper::convert_legacy(
```

Comment:
What do you think about leaving these helpers as legacy to be used as a tool when clusters, for some reason, **still** have the old legacy keys stored?